crates/worker/src/partition_processor_manager.rs

```diff
@@ -532,53 +575,105 @@ impl<T: TransportConnect> PartitionProcessorManager<T> {
+                                async move {
+                                    req.into_outgoing(Err(PartitionProcessorRpcError::NotLeader(
+                                        partition_id,
+                                    )))
                                     .send()
                                     .await
                                     .map_err(Into::into)
-                            },
-                        );
-                    }
-                    TrySendError::Closed(req) => {
-                        let _ = self.task_center.spawn(
-                            TaskKind::Disposable,
-                            "partition-processor-rpc",
-                            None,
-                            async move {
-                                req.into_outgoing(Err(PartitionProcessorRpcError::NotLeader(
-                                    partition_id,
-                                )))
-                                .send()
-                                .await
-                                .map_err(Into::into)
-                            },
-                        );
+                                },
+                            );
+                        }
                     }
                 }
             }
-        } else {
-            // ignore shutdown errors
-            let _ = self.task_center.spawn(
-                TaskKind::Disposable,
-                "partition-processor-rpc-response",
-                None,
-                async move {
-                    partition_processor_rpc
-                        .to_rpc_response(Err(PartitionProcessorRpcError::NotLeader(partition_id)))
-                        .send()
-                        .await
-                        .map_err(Into::into)
-                },
-            );
+            Some(ProcessorStatus::Starting(_)) => {
+                let _ = self.task_center.spawn(
+                    TaskKind::Disposable,
+                    "partition-processor-rpc",
+                    None,
+                    async move {
+                        partition_processor_rpc
+                            .into_outgoing(Err(PartitionProcessorRpcError::Busy))
+                            .send()
+                            .await
+                            .map_err(Into::into)
+                    },
+                );
+            }
+        }
+    }
+
+    async fn on_processor_mgr_effect(
+        &mut self,
+        effect: ProcessorEffect,
+        events: mpsc::Sender<ManagerEvent>,
+    ) -> Result<(), Error> {
+        match effect {
+            ProcessorEffect::ControlProcessor {
+                control_processor,
+                partition_table,
+            } => {
+                self.on_control_processor(control_processor, &partition_table, events)
+                    .await
+            }
+        }
+    }
+
+    fn on_processor_mgr_event(&mut self, event: ManagerEvent) {
```

Comment:
The only reason `ManagerEvent` is an enum was just to be future proof in case other type of events might arise in the future. 

This is probably already needed https://github.com/restatedev/restate/issues/2147 since one of the way to fix this is to raise an event if the partition processor returned with an error that is then handled by removing the PP entry from the Map.

Reply:
Thanks for the clarification.